src/lib/ui/thermostat/TemperatureControl.tsx

```diff
@@ -0,0 +1,126 @@
+import { type ChangeEventHandler, useEffect, useRef } from 'react'
+
+import { TemperatureAddIcon } from 'lib/icons/TemperatureAdd.js'
+import { TemperatureSubtractIcon } from 'lib/icons/TemperatureSubtract.js'
+
+const DEFAULT_MIN = 50
+const DEFAULT_MAX = 90
+
+interface TemperatureControlProps {
+  variant: 'heat' | 'cool'
+  temperature: number
+  onChange: (temperature: number) => void
+  min?: number
+  max?: number
+  temperatureUnit?: 'fahrenheit' | 'celsius'
```

Comment:
How about prefix all or none? So always note temperature

```
  temperature: number
  onChange: (temperature: number) => void
  minTemperature?: number
  minTemperature?: number
  temperatureUnit?: 'fahrenheit' | 'celsius'
```

Or simplify since temperature is implied in context

```
  value: number
  onChange: (temperature: number) => void
  min?: number
  max?: number
  unit?: 'fahrenheit' | 'celsius'
```